content/docs/reference/serializing-functions.md

```diff
@@ -6,254 +6,263 @@ The Pulumi Node.js SDK provides a core API for converting a JavaScript function
 
 ### Pulumi Execution Order
 
-`pulumi` uses `node` to execute a Pulumi application.  During execution, when a call to `createLambdaFunction` is encountered, the function is converted to a Lambda at that point in execution.  That means that if the function captures any state, then the value that is captured will be whatever it was at the point in time.
+`pulumi` uses `node` to execute a Pulumi application. During execution, when a call to `createLambdaFunction` is encountered, the function is converted to a Lambda at that point in execution. That means that if the function captures any state, then the value that is captured will be whatever it was at the point in time.
 
-For this reason, it is highly recommended that code not capture values which are also mutated in code.  It is much safer and easier to reason about immutable values that are captured in code.
+For this reason, it is highly recommended that code not capture values which are also mutated in code. It is much safer and easier to reason about immutable values that are captured in code.
 
 To see the problems this avoids in practice, consider the following two programs:
 
 ```typescript
 let obj = { a: 1, b: 2 };
 
-const lambda: aws.lambda.Function = serverless.function.createLambdaFunction("mylambda", 
-    () => {
-        console.log(obj);
-    });
+const lambda = new aws.lambda.CallbackFunction("mylambda", {
+  callback: async () => {
+    console.log(obj);
+  }
+});
 
 obj = { a: 3, b: 4 };
 ```
 
-
 ```typescript
 let obj = { a: 1, b: 2 };
 obj = { a: 3, b: 4 };
 
-const lambda: aws.lambda.Function = serverless.function.createLambdaFunction("mylambda", 
-    () => {
-        console.log(obj);
-    });
+const lambda = new aws.lambda.CallbackFunction("mylambda", {
+  callback: async () => {
+    console.log(obj);
+  }
+});
 ```
 
-When `pulumi` starts executing `createLambdaFunction` it will analyze the JavaScript function code and will see that it uses the 'obj' value.  At that point in time it will use whatever the value is currently to serialize over.  So, in the first example, it will serialize the value `{ a: 1, b: 2 }`, even though right after executing createLambdaFunction the program code will update that value to `{ a: 3, b: 4}`.  In the second example, the code will see the `{ a: 3, b: 4 }` value and will serialize that into the 'run time' code.  
+When `pulumi` starts executing `createLambdaFunction` it will analyze the JavaScript function code and will see that it uses the 'obj' value. At that point in time it will use whatever the value is currently to serialize over. So, in the first example, it will serialize the value `{ a: 1, b: 2 }`, even though right after executing createLambdaFunction the program code will update that value to `{ a: 3, b: 4}`. In the second example, the code will see the `{ a: 3, b: 4 }` value and will serialize that into the 'run time' code.
```

Comment:
```suggestion
When `pulumi` starts executing `new aws.lambda.CallbackFunction` it will analyze the JavaScript function code and will see that it uses the 'obj' value. At that point in time it will use whatever the value is currently to serialize over. So, in the first example, it will serialize the value `{ a: 1, b: 2 }`, even though right after executing createLambdaFunction the program code will update that value to `{ a: 3, b: 4}`. In the second example, the code will see the `{ a: 3, b: 4 }` value and will serialize that into the 'run time' code.
```